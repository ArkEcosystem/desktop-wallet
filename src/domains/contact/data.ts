export const contact1 = {
	name: () => "ROBank",
	addresses: () => [
		{
			coin: "Ark",
			network: "mainnet",
			address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
			avatar: "avatar",
			isDelegate: () => true,
			isBusiness: () => true,
			isBridgechain: () => true,
		},
	],
};

export const contact2 = {
	name: () => "OLEBank",
	addresses: () => [
		{
			coin: "Ark",
			network: "mainnet",
			address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
			avatar: "avatar",
			isDelegate: () => true,
			isBusiness: () => false,
			isBridgechain: () => false,
		},
		{
			coin: "Bitcoin",
			network: "mainnet",
			address: "15pyr1HRAxpq3x64duXav1csmyCtXXu9G8",
			avatar: "avatar",
			isDelegate: () => false,
			isBusiness: () => false,
			isBridgechain: () => false,
		},
		{
			coin: "Ethereum",
			network: "mainnet",
			address: "0x5e8f7a63e31c759ef0ad5e71594e838b380d7c33",
			avatar: "avatar",
			isDelegate: () => false,
			isBusiness: () => true,
			isBridgechain: () => false,
		},
	],
};

export const contacts = [contact1, contact2];

export const networks = [
	{
		icon: "Ark",
		name: "ARK Ecosystem",
		className: "text-theme-danger-400 border-theme-danger-200",
	},
	{
		icon: "Bitcoin",
		name: "Bitcoin",
		className: "text-theme-warning-400 border-theme-warning-200",
	},
	{
		icon: "Ethereum",
		name: "Ethereum",
		className: "text-theme-neutral-800 border-theme-neutral-600",
	},
	{
		icon: "Lisk",
		name: "Lisk",
		className: "text-theme-primary-600 border-theme-primary-400",
	},
	{
		icon: "Ripple",
		name: "Ripple",
		className: "text-theme-primary-700 border-theme-primary-500",
	},
];
