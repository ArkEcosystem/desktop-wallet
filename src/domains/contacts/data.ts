export const contact1 = {
  name: () => "ROBank",
  addresses: () => [
    {
      coin: "Ark",
      network: "mainnet",
      address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
      isDelegate: () => true,
      isBusiness: () => true,
      isBridgechain: () => true,
    },
  ],
};

export const contact2 = {
  name: () => "OLEBank",
  addresses: () => [
    {
      coin: "Ark",
      network: "mainnet",
      address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
      isDelegate: () => true,
      isBusiness: () => false,
      isBridgechain: () => false,
    },
    {
      coin: "Bitcoin",
      network: "mainnet",
      address: "15pyr1HRAxpq3x64duXav1csmyCtXXu9G8",
      isDelegate: () => false,
      isBusiness: () => false,
      isBridgechain: () => false,
    },
    {
      coin: "Ethereum",
      network: "mainnet",
      address: "0x5e8f7a63e31c759ef0ad5e71594e838b380d7c33",
      isDelegate: () => false,
      isBusiness: () => true,
      isBridgechain: () => false,
    },
  ],
};

export const contacts = [
  contact1,
  contact2,
];
