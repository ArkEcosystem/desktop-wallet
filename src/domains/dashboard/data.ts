export const networks = [
	{
		name: "Ark",
		isSelected: true,
	},
	{
		name: "Ethereum",
		isSelected: true,
	},
	{
		name: "Bitcoin",
		isSelected: false,
	},
];

export const wallets = [
	{
		coinIcon: "Ark",
		coinClass: "text-theme-danger-400 border-theme-danger-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "2,000,000 ARK",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Ethereum",
		coinClass: "text-theme-neutral-800 border-theme-neutral-600",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Eth",
		balance: "100,150 ETH",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Ark",
		coinClass: "text-theme-danger-400 border-theme-danger-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "2,000,000 ARK",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Ethereum",
		coinClass: "text-theme-neutral-800 border-theme-neutral-600",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Eth",
		balance: "100,150 ETH",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		isBlank: true,
		blankTitle: "You have 8 more wallets",
		blankTitleClass: "text-theme-neutral-700 cursor-pointer",
		blankSubtitle: "Look all wallet",
		blankSubtitleClass: "text-theme-primary-600 font-semibold cursor-pointer",
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
	{
		coinIcon: "Bitcoin",
		coinClass: "text-theme-warning-400 border-theme-warning-200",
		avatarId: "test",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		balance: "100 BTC",
		fiat: "1,000,000 USD",
		walletTypeIcons: ["Star", "Multisig", "Ledger"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
	},
];

// Transaction History data
export const transactions = [
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "receive",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "100 BTC",
		fiat: "1,000,000 USD",
		transactionInfoIcons: ["Multisig", "Ledger"],
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "send",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "- 100 BTC",
		fiat: "1,000,000 USD",
		transactionInfoIcons: ["Multisig", "Ledger"],
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "receive",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "100 BTC",
		fiat: "1,000,000 USD",
		transactionInfoIcons: ["Multisig", "Ledger"],
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "send",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "- 100 BTC",
		fiat: "1,000,000 USD",
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "receive",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "100 BTC",
		fiat: "1,000,000 USD",
		transactionInfoIcons: ["Ledger"],
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "send",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "- 100 BTC",
		fiat: "1,000,000 USD",
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "receive",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "100 BTC",
		fiat: "1,000,000 USD",
	},
	{
		date: "17 Mar 2020 22:02:10",
		avatarId: "test",
		type: "send",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "My Wallet",
		amount: "- 100 BTC",
		fiat: "1,000,000 USD",
	},
];

export const portfolioPercentages = [
	{
		label: "ARK",
		value: 40,
		color: "danger-500",
	},
	{
		label: "BTC",
		value: 25,
		color: "warning-500",
	},
	{
		label: "ETH",
		value: 20,
		color: "primary-500",
	},
	{
		label: "Other",
		value: 15,
		color: "neutral-400",
	},
];

export const balances = [
	{ name: "Mon", label: "22 Jun, 2020", usd: 1000, btc: 2400, formatted: { usd: "1,000", btc: "0.26" } },
	{ name: "Tue", label: "23 Jun, 2020", usd: 3000, btc: 5400, formatted: { usd: "3,000", btc: "0.58" } },
	{ name: "Wed", label: "24 Jun, 2020", usd: 15100, btc: 1398, formatted: { usd: "15,100", btc: "0.15" } },
	{ name: "Thu", label: "25 Jun, 2020", usd: 4000, btc: 9800, formatted: { usd: "4,000", btc: "1.06" } },
	{ name: "Fri", label: "26 Jun, 2020", usd: 11200, btc: 2000, formatted: { usd: "11,200", btc: "0.22" } },
	{ name: "Sat", label: "27 Jun, 2020", usd: 3000, btc: 4800, formatted: { usd: "3,000", btc: "0.52" } },
	{ name: "Sun", label: "28 Jun, 2020", usd: 8000, btc: 18000, formatted: { usd: "8,000", btc: "2.16" } },
];
