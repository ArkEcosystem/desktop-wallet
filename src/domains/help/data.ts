export const categories = [
	{
		title: "Profile",
		subtitle: "38 topics",
		icon: "Profile",
		path: "/",
	},
	{
		title: "Portfolio",
		subtitle: "9 topics",
		icon: "Portfolio",
		path: "/",
	},
	{
		title: "Plugins",
		subtitle: "106 topics",
		icon: "Plugins",
		path: "/",
	},
	{
		title: "Exchanges",
		subtitle: "28 topics",
		icon: "Exchange",
		path: "/",
	},
];

export const searchData = [
	{
		title: "How do I install the plugin?",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
	{
		title: "Installing the side plugin",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
	{
		title: "If the plugin is not installed, what do we do?",
		category: "Plugin",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as the bla bla bla",
	},
];

export const helpfulArticles = [
	{
		title: "How to register an account?",
		path: "/",
	},
	{
		title: "How to enter your account",
		path: "/",
	},
	{
		title:
			"How can I change my personal data if I do not have access to my account (forgot my password) and to my mail?",
		path: "/",
	},
	{
		title:
			"If you need to make changes to the data entered during the passage of verification, where to apply, how to issue a request?",
		path: "/",
	},
	{
		title: "Can I delete my account",
		path: "/",
	},
];

export const popularArticles = [
	{
		title: "How to delete YubiKey for Two-factor Authentication (2FA)",
		path: "/",
	},
	{
		title: "Using a YubiKey for Two-Factor Authentication (2FA)",
		path: "/",
	},
	{
		title: "2FA Code Error",
		path: "/",
	},
	{
		title: "How to Reset SMS Authentication",
		path: "/",
	},
	{
		title: "Supported SMS countries Not Receiving SMS Auth Codes",
		path: "/",
	},
];

export const newestArticles = [
	{
		title: "How to withdraw from wallet",
		path: "/",
	},
	{
		title: "How to Make Internal Transfer within wallet",
		path: "/",
	},
	{
		title: "Missing Withdrawals",
		path: "/",
	},
	{
		title: "Withdrawal to a Wrong Address",
		path: "/",
	},
	{
		title: "How to Resume the Withdrawal Function",
		path: "/",
	},
];

export const faqArticles = [
	{
		title: "Development Environment Setup",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
	{
		title: "Core-Tester provides set of instructions on how to run commands for creating and sending transactions?",
		description:
			"With our testnet up and running, the first thing we need to do is to connect to it. All interactions between ARK Nodes and the outside world happen through the Public API, which is a REST API facilitating different actions on the blockchain.",
		path: "/",
	},
	{
		title: "Setup Your Own Blockchain Explorer",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
	{
		title: "Is it essential for you to understand and master the art of ARK Core Module development?",
		description:
			"ARK Core is written in TypeScript, and it has been using Lerna to manage the development and publication of its packages and uses Node.js as execution environment. This guid will take you through the basic steps of setting up a development environment",
		path: "/",
	},
];

export const article = {
	title: "Setup Your Own Blockchain Explorer",
	body: "jfdskfjdskajfkasjfasdf",
	category: "Portfolio",
	categoryIcon: "PortfolioTag",
	views: "32 908",
	image: "https://picsum.photos/seed/picsum/800/400",
	sections: [
		{
			title: "1. Introduction",
			id: "introduction",
			body:
				"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.\n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.",
		},
		{
			title: "2. Create Account",
			id: "account",
			body:
				"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.\n\n Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Tempus imperdiet nulla malesuada pellentesque elit eget. Bibendum arcu vitae elementum curabitur. Eget lorem dolor sed viverra ipsum. Ut venenatis tellus in metus vulputate eu scelerisque. Mi bibendum neque egestas congue quisque egestas diam in. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper malesuada proin. Nullam vehicula ipsum a arcu cursus. Sed libero enim sed faucibus. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Quis imperdiet massa tincidunt nunc pulvinar sapien et ligula. Dui id ornare arcu odio ut sem nulla pharetra diam. Diam quam nulla porttitor massa.",
		},
	],
};
