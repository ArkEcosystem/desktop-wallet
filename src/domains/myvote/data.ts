export const addressListData = [
	{
		walletAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		walletName: "OLEBank",
		type: "Multisig",
		balance: "432,509.08946384 ARK",
		delegateAddress: "DACFobAjNEKsc1CtPSMyp5uA4wp6uC3fgC",
		delegateName: "Delegate 1",
		rank: 2,
		msqUrl: "https://marketsquare.ark.io",
		isActive: true,
	},
	{
		walletAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		type: "Ledger",
		balance: "8,523.3884504 ARK",
		delegateAddress: "DBozeHpqr7fie1QrfmNcGzdtEb7NB73AGq",
		delegateName: "Delegate 45",
		rank: 45,
		isActive: true,
	},
	{
		walletAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		balance: "0.000462 ARK",
	},
	{
		walletAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		balance: "0.00 ARK",
	},
];

export const delegateListData = [
	{
		delegateAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		delegateName: "Delegate 1",
		rank: 1,
		votes: 3.43,
		msqUrl: "https://marketsquare.ark.io",
		commissionPercentage: 80,
		commissionDaily: 2.387082496,
		payout: "Daily",
		min: 500,
	},
	{
		delegateAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		delegateName: "Delegate 2",
		rank: 2,
		votes: 3.43,
		msqUrl: "https://marketsquare.ark.io",
		commissionPercentage: 80,
		commissionDaily: 2.387082496,
		payout: "Daily",
		min: 500,
	},
	{
		delegateAddress: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		delegateName: "Delegate 3",
		rank: 3,
		votes: 3.43,
	},
];
