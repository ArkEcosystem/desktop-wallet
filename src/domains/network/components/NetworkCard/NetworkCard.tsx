import { CardControl } from "app/components/Card";
import { Circle } from "app/components/Circle";
import { Icon } from "app/components/Icon";
import React from "react";

type NetworkCardProps = {
	icon?: string;
	name?: string;
	network?: string;
	onChange?: any;
};

export const NetworkCard = ({ icon, name, network, onChange }: NetworkCardProps) => (
	<CardControl type="radio" value={name} name="network" onChange={() => onChange(name)}>
		<div className="flex items-center py-2">
			<Circle className="transition-colors duration-100 border-theme-neutral-300" size="lg" noShadow>
				<Icon name={icon!} />
			</Circle>
			<div className="flex flex-col ml-4">
				<span className="text-sm font-semibold text-theme-neutral-500" data-testid="network-card--network">
					{network}
				</span>
				<span className="font-semibold text-theme-text" data-testid="network-card--name">
					{name}
				</span>
			</div>
		</div>
	</CardControl>
);

NetworkCard.defaultProps = {
	network: "Mainnet",
};
