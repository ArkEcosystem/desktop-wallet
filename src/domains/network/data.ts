import { ARK } from "@arkecosystem/platform-sdk-ark";
import { BTC } from "@arkecosystem/platform-sdk-btc";
import { LSK } from "@arkecosystem/platform-sdk-lsk";
import { NetworkData } from "@arkecosystem/platform-sdk-profiles";
import { XRP } from "@arkecosystem/platform-sdk-xrp";

export interface CoinNetworkExtended {
	displayName: string;
	borderClass: string;
	textClass: string;
	iconName: string;
}

export const coinsNetworkExtendedData: Record<string, Record<string, CoinNetworkExtended>> = {
	ARK: {
		Mainnet: {
			displayName: "Ark",
			borderClass: "border-theme-danger-light",
			textClass: "text-theme-danger-400",
			iconName: "Ark",
		},
		Devnet: {
			displayName: "Ark Devnet",
			borderClass: "border-theme-primary-100",
			textClass: "text-theme-primary-400",
			iconName: "Ark",
		},
	},
	BTC: {
		Livenet: {
			displayName: "Bitcoin",
			borderClass: "border-theme-warning-200",
			textClass: "text-theme-warning-400",
			iconName: "Bitcoin",
		},
	},
	LSK: {
		Mainnet: {
			displayName: "Lisk",
			borderClass: "border-theme-primary-400",
			textClass: "text-theme-primary",
			iconName: "Lisk",
		},
	},
	XRP: {
		Mainnet: {
			displayName: "Ripple",
			borderClass: "border-theme-primary-500",
			textClass: "text-theme-primary-dark",
			iconName: "Ripple",
		},
	},
};

export const availableNetworksMock: NetworkData[] = [
	new NetworkData(ARK.manifest.name, ARK.manifest.networks.mainnet),
	new NetworkData(ARK.manifest.name, ARK.manifest.networks.devnet),
	new NetworkData(BTC.manifest.name, BTC.manifest.networks.livenet),
	new NetworkData(LSK.manifest.name, LSK.manifest.networks.mainnet),
	new NetworkData(XRP.manifest.name, XRP.manifest.networks.mainnet),
];
