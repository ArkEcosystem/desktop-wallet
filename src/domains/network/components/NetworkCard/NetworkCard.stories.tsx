import React from "react";

import { networks } from "../../data";
import { NetworkCard } from "./NetworkCard";

export default { title: "Network / Components / Network Card" };

export const Default = () => <NetworkCard icon="Ark" name="ARK Ecosystem" network="Mainnet" />;

export const NetworksList = () => {
	return (
		<div className="grid grid-flow-col grid-cols-4 gap-2">
			{networks.map((network) => (
				<NetworkCard key={network.name} icon={network.icon} name={network.name} network={network.network} />
			))}
		</div>
	);
};
