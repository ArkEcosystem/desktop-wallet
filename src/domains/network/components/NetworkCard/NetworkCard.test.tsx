/* eslint-disable @typescript-eslint/require-await */
import React from "react";
import { act, fireEvent, render } from "testing-library";

import { networks } from "../../data";
import { NetworkCard } from "./NetworkCard";

describe("NetworkCard", () => {
	const network = networks[0];

	it("should render", () => {
		const { container, asFragment, getByTestId } = render(<NetworkCard {...network} />);

		expect(container).toBeTruthy();
		expect(getByTestId("network-card--network")).toHaveTextContent(network.network);
		expect(getByTestId("network-card--name")).toHaveTextContent(network.name);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should select a network card", async () => {
		const onChange = jest.fn();
		const { asFragment, getAllByTestId } = render(<NetworkCard {...network} onChange={onChange} />);

		expect(asFragment()).toMatchSnapshot();

		const networksEl = getAllByTestId(`card-control__network`);
		expect(networksEl.length).toEqual(1);

		await act(async () => {
			fireEvent.click(networksEl[0]);
		});

		expect(onChange).toBeCalledWith(network.name);
	});
});
