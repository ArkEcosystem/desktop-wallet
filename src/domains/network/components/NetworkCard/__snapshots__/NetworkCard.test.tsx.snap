// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkCard should render 1`] = `
<DocumentFragment>
  <label
    class="CardControl___StyledLabel-sc-28luv0-3 PNVuh"
  >
    <input
      class="CardControl__Input-sc-28luv0-0 lbnjeC"
      data-testid="card-control__network"
      name="network"
      type="radio"
      value="ARK Ecosystem"
    />
    <div
      class="sc-AxjAm eWJHeF CardControl__CustomCard-sc-28luv0-1 dojmPD"
    >
      <div
        class="flex items-center py-2"
      >
        <div
          class="sc-AxiKw gDUlrm transition-colors duration-100 border-theme-neutral-300"
        >
          <div
            class="sc-AxirZ iQaTP"
            height="1em"
            width="1em"
          >
            <svg>
              ark.svg
            </svg>
          </div>
        </div>
        <div
          class="flex flex-col ml-4"
        >
          <span
            class="text-sm font-semibold text-theme-neutral-500"
            data-testid="network-card--network"
          >
            Mainnet
          </span>
          <span
            class="font-semibold text-theme-text"
            data-testid="network-card--name"
          >
            ARK Ecosystem
          </span>
        </div>
      </div>
    </div>
  </label>
</DocumentFragment>
`;

exports[`NetworkCard should select a network card 1`] = `
<DocumentFragment>
  <label
    class="CardControl___StyledLabel-sc-28luv0-3 PNVuh"
  >
    <input
      class="CardControl__Input-sc-28luv0-0 lbnjeC"
      data-testid="card-control__network"
      name="network"
      type="radio"
      value="ARK Ecosystem"
    />
    <div
      class="sc-AxjAm eWJHeF CardControl__CustomCard-sc-28luv0-1 dojmPD"
    >
      <div
        class="flex items-center py-2"
      >
        <div
          class="sc-AxiKw gDUlrm transition-colors duration-100 border-theme-neutral-300"
        >
          <div
            class="sc-AxirZ iQaTP"
            height="1em"
            width="1em"
          >
            <svg>
              ark.svg
            </svg>
          </div>
        </div>
        <div
          class="flex flex-col ml-4"
        >
          <span
            class="text-sm font-semibold text-theme-neutral-500"
            data-testid="network-card--network"
          >
            Mainnet
          </span>
          <span
            class="font-semibold text-theme-text"
            data-testid="network-card--name"
          >
            ARK Ecosystem
          </span>
        </div>
      </div>
    </div>
  </label>
</DocumentFragment>
`;
